src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -492,11 +498,11 @@ ss::future<> async_manifest_view::run_bg_loop() {
 ss::future<result<std::unique_ptr<async_manifest_view_cursor>, error_outcome>>
 async_manifest_view::get_cursor(
   async_view_search_query_t query,
-  std::optional<model::offset> end_inclusive) noexcept {
+  std::optional<cursor_range> cursor_range) noexcept {
```

Comment:
The api looks confusing to me, we already have a query parameter which indicates a starting point, but we also have an optional range parameter with optional `begin_inclusive` field. Can we just use `query` parameter as `begin_inclusive`? The whole point of the cursor is to read segments metadata from the point A to the point B.

I understand that it was used to extend the cursor to the area between the clean and start offsets. But I think that this could be done differently allowing offsets in the clean-start range to be fetched by the cursor. Currently, you can only create the cursor that points to something in archive-start/stm-start range or in the STM offset range. You can extend it to archive-clean/start range. This `range` parameter is only used once with `archive_clean_offset` as `begin_inclusive`. So why do we need to make it too generic?

Reply:
I agree. It wasn't great. I've updated the code to allow `get_cursor` to operate between the clean and start offsets of the archive as you suggested.